File: app/src/main/java/com/fonfon/noloss/lib/ObservableUtils.java
package com.fonfon.noloss.lib;

import android.os.Looper;

import io.reactivex.Observer;

public final class ObservableUtils {

  public static boolean checkMainThread(Observer<?> observer) {
    if (Looper.myLooper() != Looper.getMainLooper()) {
      observer.onError(new IllegalStateException(
          "Expected to be called on the main thread but was " + Thread.currentThread().getName()));
      return false;
    }
    return true;
  }
}

File: app/src/main/java/com/fonfon/noloss/lib/ObservableUtils.kt
package com.fonfon.noloss.lib

import android.os.Looper

import io.reactivex.Observer

object ObservableUtils {

  fun checkMainThread(observer: Observer<*>): Boolean {
    if (Looper.myLooper() != Looper.getMainLooper()) {
      observer.onError(IllegalStateException("""Expected to be called on the main thread but was ${Thread.currentThread().name}"""))
      return false
    }
    return true
  }
}
